worker/controlsocket/handlers.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package controlsocket
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/environs/bootstrap"
+	"github.com/juju/juju/state"
+)
+
+const (
+	// jujuMetricsUserPrefix defines the "namespace" in which this worker is
+	// allowed to create/remove users.
+	jujuMetricsUserPrefix = "juju-metrics-"
+
+	// userCreator is the listed "creator" of metrics users in state.
+	userCreator = "admin"
+)
+
+func (w *Worker) registerHandlers(handle func(pattern string, handler http.Handler)) {
+	// TODO: does this API structure make sense?
+	handle("/metrics-users/add", http.HandlerFunc(w.handleAddMetricsUser))
+	handle("/metrics-users/remove", http.HandlerFunc(w.handleRemoveMetricsUser))
+}
+
+func (w *Worker) handleAddMetricsUser(resp http.ResponseWriter, req *http.Request) {
```

Comment:
Validate the method is correct. I do agree with @jnsgruk that it would be better if `GET` and `DELETE` methods.

Reply:
Also what happens if you attempt to add the user again, it will return InternalServerError. I suspect you want that to be idempotent as well.